apps/designer/app/shared/dom-hooks/use-measure.ts

```diff
@@ -28,6 +28,18 @@ export const useMeasure = <
     onScrollEnd: handleChange,
   });
 
+  // Detect movement of the element without remounting.
+  useEffect(() => {
```

Comment:
I think this will not disconnect the observer if element ref changes, that's why for resize observer, observer is memorized separately, so we create only one observer and then separately call .observe(